src/sentry/profiles/utils.py

```diff
@@ -171,3 +176,24 @@ def parse_profile_filters(query: str) -> dict[str, str]:
         profile_filters[term.key.name] = term.value.value
 
     return profile_filters
+
+
+def apply_stack_trace_rules_to_profile(profile: Profile, rules_config: str) -> None:
+    profiling_rules = keep_profiling_rules(rules_config)
+    if profiling_rules == "":
+        return
+    enhancements = Enhancements.from_config_string(profiling_rules)
+    if "version" in profile:
+        enhancements.apply_modifications_to_frame(
+            profile["profile"]["frames"], profile["platform"], {}
+        )
+    elif profile["platform"] == "android":
+        # set the fields that Enhancements expect
+        # with the right names.
+        for method in profile["profile"]["methods"]:
+            method["function"] = method.get("name", "")
+            method["abs_path"] = method.get("source_file", "")
+            method["module"] = method.get("class_name", "")
+        enhancements.apply_modifications_to_frame(
+            profile["profile"]["methods"], profile["platform"], {}
+        )
```

Comment:
The only action this is capable of handling is making a frame as in app or not right? Can we make that clear somehow? Because it's not immediately obvious why setting these aliases will work.

Reply:
Yeah, I can add a few line of comments to clarify which actions and matchers supported (only a subset).

As for why we need the aliases, it's simply because the `Enhancer` expects a list of frames that have those fields, which we have for the sample format but not for the android format.